plutus-ledger-api/test/Spec/CostModelParams.hs

```diff
@@ -25,39 +26,43 @@ tests =
   testGroup
     "costModelParams"
     [ testCase "length" $ do
-            166 @=? length (enumerate @V1.ParamName)
-            175 @=? length (enumerate @V2.ParamName)
-            223 @=? length (enumerate @V3.ParamName)
-            -- FIXME: the following three tests are misleading; better to directly test that
-            -- their `showParamName` implemementation is injective
-            166 @=? length v1CostModelParamNames
-            175 @=? length v2CostModelParamNames
-            223 @=? length v3CostModelParamNames
+            166 @=? length v1_ParamNames
+            166 @=? length v1_costModelParamsForTesting
+            175 @=? length v2_ParamNames
+            175 @=? length v2_costModelParamsForTesting
+            223 @=? length v3_ParamNames
+            223 @=? length v3_costModelParamsForTesting
+    , testCase "tripping paramname" $ do
+            for_ v1_ParamNames $ \ p ->
+                Just p @=? readParamName (showParamName p)
+            for_ v2_ParamNames $ \ p ->
+                Just p @=? readParamName (showParamName p)
+            for_ v3_ParamNames $ \ p ->
+                Just p @=? readParamName (showParamName p)
+    , testCase "default values costmodelparamsfortesting" $ do
+            defaultCostModelParams @=? Just (toCostModelParams v3_costModelParamsForTesting)
     , testCase "context length" $ do
-            let costValuesForTesting = Map.elems costModelParamsForTesting
+            let costValuesForTesting = fmap snd v3_costModelParamsForTesting
             -- the `costModelParamsForTesting` reflects only the latest version V3, so this should succeed because the lengths match
             assertBool "wrong number of arguments in V3.mkEvaluationContext" $ isRight $ runExcept $ runWriterT $ V3.mkEvaluationContext costValuesForTesting
             -- this one should succeed because we pass more params
-            assertBool "larger number of params did not warn" $ hasWarnMoreParams (length v3CostModelParamNames) (length v3CostModelParamNames+1) $
+            assertBool "larger number of params did not warn" $ hasWarnMoreParams (length v3_ParamNames) (1 + length v3_ParamNames) $
```

Comment:
I've forgotten why we allow this.  Is it written down somewhere and if so could we have a reference here?

Reply:
> Is it written down somewhere

Ah, I found it in `Note [Cost model parameters from the ledger's point of view]`.